blueocean-dashboard/src/main/js/components/RunDetails.jsx

```diff
@@ -166,31 +169,22 @@ class RunDetails extends Component {
 
         const base = { base: baseUrl };
 
-        const failureCount = Math.min(99, (testSummary && parseInt(testSummary.failed)) || 0);
-        const testsBadge = failureCount > 0 && <div className="TabBadgeIcon">{failureCount}</div>;
-
         const tabs = [
             <TabLink to="/pipeline" {...base}>
                 {t('rundetail.header.tab.pipeline', {
                     defaultValue: 'Pipeline',
                 })}
             </TabLink>,
-            <TabLink to="/changes" {...base}>
-                {t('rundetail.header.tab.changes', {
-                    defaultValue: 'Changes',
-                })}
-            </TabLink>,
-            <TabLink to="/tests" {...base}>
-                {t('rundetail.header.tab.tests', { defaultValue: 'Tests' })}
-                {testsBadge}
-            </TabLink>,
-            <TabLink to="/artifacts" {...base}>
-                {t('rundetail.header.tab.artifacts', {
-                    defaultValue: 'Artifacts',
-                })}
-            </TabLink>,
         ];
 
+        actions.map(descriptor => {
+            const action = new descriptor({pipeline, run: currentRun});
+            const badge = action.notification && <div className="TabBadgeIcon">{ action.notification }</div>;
+            tabs.push(
+                <TabLink to={'/' + action.name} { ...base }>{action.title}{badge}</TabLink>
```

Comment:
Create a `TabLink` for each registered extension of the `jenkins.run.actions` extension point. Previously, the default run details actions (changes, tests, and artifacts) had hard-coded `TabLink` components.